homeassistant/components/motion_blinds/cover.py

```diff
@@ -0,0 +1,259 @@
+    BlindType.Curtain: DEVICE_CLASS_CURTAIN,
+    BlindType.CurtainLeft: DEVICE_CLASS_CURTAIN,
+    BlindType.CurtainRight: DEVICE_CLASS_CURTAIN,
+}
+
+TILT_DEVICE_MAP = {
+    BlindType.VenetianBlind: DEVICE_CLASS_BLIND,
+    BlindType.ShangriLaBlind: DEVICE_CLASS_BLIND,
+    BlindType.DoubleRoller: DEVICE_CLASS_SHADE,
+}
+
+TDBU_DEVICE_MAP = {
+    BlindType.TopDownBottomUp: DEVICE_CLASS_SHADE,
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Motion Blind from a config entry."""
+    entities = []
+    motion_gateway = hass.data[DOMAIN][config_entry.entry_id][KEY_GATEWAY]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
+
+    for blind in motion_gateway.device_list.values():
+        if blind.type in POSITION_DEVICE_MAP:
+            entities.append(
+                MotionPositionDevice(
+                    coordinator, blind, POSITION_DEVICE_MAP[blind.type], config_entry
+                )
+            )
+
+        elif blind.type in TILT_DEVICE_MAP:
+            entities.append(
+                MotionTiltDevice(
+                    coordinator, blind, TILT_DEVICE_MAP[blind.type], config_entry
+                )
+            )
+
+        elif blind.type in TDBU_DEVICE_MAP:
+            entities.append(
+                MotionTDBUDevice(
+                    coordinator, blind, TDBU_DEVICE_MAP[blind.type], config_entry, "Top"
+                )
+            )
+            entities.append(
+                MotionTDBUDevice(
+                    coordinator,
+                    blind,
+                    TDBU_DEVICE_MAP[blind.type],
+                    config_entry,
+                    "Bottom",
+                )
+            )
+
+        else:
+            _LOGGER.warning("Blind type '%s' not yet supported", blind.blind_type)
+
+    async_add_entities(entities)
+
+
+class MotionPositionDevice(CoordinatorEntity, CoverEntity):
+    """Representation of a Motion Blind Device."""
+
+    def __init__(self, coordinator, blind, device_class, config_entry):
+        """Initialize the blind."""
+        super().__init__(coordinator)
+
+        self._blind = blind
+        self._device_class = device_class
+        self._config_entry = config_entry
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the blind."""
+        return self._blind.mac
+
+    @property
+    def device_info(self):
+        """Return the device info of the blind."""
+        device_info = {
+            "identifiers": {(DOMAIN, self._blind.mac)},
+            "manufacturer": MANUFACTURER,
+            "name": f"{self._blind.blind_type}-{self._blind.mac}",
+            "model": self._blind.blind_type,
+            "via_device": (DOMAIN, self._config_entry.unique_id),
+        }
+
+        return device_info
+
+    @property
+    def name(self):
+        """Return the name of the blind."""
+        return f"{self._blind.blind_type}-{self._blind.mac}"
```

Comment:
Is there a more human readable name that can be used?

Reply:
Not really, the self._blind.blind_type is human readable like "RollerBlind" or "DoubleRoller", but unfortunately the API does not allow me to fetch the names set in the Mobile Motion Application. In order to distinguis two blinds with the same type, the only thing available is the mac adress of the blind...

Reply:
Maybe only take the last 6 characters of the mac?

Reply:
that actually makes sence, the last 4 characters of the mac of a blind are actually indicating the blind and the first 12 characters are always the mac of the gateway to which the blind is connected.

So taking only the last 4 characters makes a lot of sence!
I am making the changes right now.

Reply:
Done, just pushed the commit